src/test/java/uk/ac/ebi/eva/pipeline/ApplicationTest.java

```diff
@@ -0,0 +1,56 @@
+package uk.ac.ebi.eva.pipeline;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.batch.core.ExitStatus;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobInstance;
+import org.springframework.batch.core.explore.JobExplorer;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import uk.ac.ebi.eva.pipeline.configuration.JobOptions;
+import uk.ac.ebi.eva.pipeline.jobs.GenotypedVcfJob;
+import uk.ac.ebi.eva.test.rules.TemporaryMongoRule;
+
+import java.util.List;
+
+/**
+ * The purpose of this test is to imitate an execution made by an user through the CLI.
+ * This is needed because all the other tests just instantiate what they need (just one step, or just one job) and
+ * sometimes we have errors due to collisions instantiating several jobs. This test should instantiate everything
+ * Spring instantiates in a real execution.
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@ActiveProfiles({"integrationTest,test,mongo"})
+public class ApplicationTest {
```

Comment:
here is the first version that didn't check if the job was completed or failed https://github.com/jmmut/eva-pipeline/blob/17b48eb4746b54b5bcec0d42f6842803bc64fcca/src/test/java/uk/ac/ebi/eva/pipeline/ApplicationTest.java